crates/shared/src/sources.rs

```diff
@@ -7,22 +7,17 @@ pub mod uniswap_v3;
 pub mod uniswap_v3_pair_provider;
 
 use {
-    self::uniswap_v2::{
-        pair_provider::PairProvider,
-        pool_fetching::{Pool, PoolFetching},
-    },
-    crate::{ethrpc::Web3, recent_block_cache::Block},
+    self::uniswap_v2::pool_fetching::{Pool, PoolFetching},
+    crate::recent_block_cache::Block,
     anyhow::{bail, Result},
     model::TokenPair,
-    std::{
-        collections::{HashMap, HashSet},
-        sync::Arc,
-    },
+    std::{collections::HashSet, sync::Arc},
 };
 
 #[derive(Debug, Clone, Copy, Eq, Hash, PartialEq, clap::ValueEnum)]
 #[clap(rename_all = "verbatim")]
 pub enum BaselineSource {
+    None,
```

Comment:
Why would we need a `None` variant here? Shouldn't an optional `BaselineSource` rather be `Option<BaselineSource>`?

Reply:
I forgot to explain this in the PR description. The problem is that in our Clap arguments we specify baselines sources as `Option<Vec<BaselineSource>>`. I have not found a way to have this parsed as `Some(vec![])` to indicate that we don't want any hardcoded baseline source.

Reply:
I remember running into the same [issue](https://github.com/cowprotocol/services/issues/901). I did a bit of research just now and all the documentation suggests that it should work out of the box (which it apparently doesn't).
But I was able to make it work by specifying `num_args = 0..` in the clap derive macro:
```rust
    /// Which Liquidity sources to be used by Price Estimator.
    #[clap(
        long,
        env,
        ignore_case = true,
        use_value_delimiter = true,
        num_args = 0..
    )]
    pub baseline_sources: Option<Vec<BaselineSource>>,
```

```
cargo run -p solver -- --baseline-sources
...
base_tokens: []
baseline_sources: Some([])
pool_cache_blocks: 10
...
```

```
cargo run -p solver -- --baseline-sources "UniswapV2"
...
base_tokens: []
baseline_sources: Some([UniswapV2])
pool_cache_blocks: 10
...
```

Reply:
I just remembered that we configure everything with env variables and with those I wasn't able to make it work yet. 🤔

Reply:
>  I have not found a way to have this parsed as Some([]) to indicate that we don't want any hardcoded baseline source.

You can't. It's annoying. I opened an issue/discussion in the past, but never got anywhere.

Reply:
> You can't. It's annoying.

@nlordell  Did you read my comment? It's certainly possible. Just not with environment variables.

Reply:
I did! Sorry for the short reply, what I meant was to confirming that I came to the same conclusion a while ago (can't specify an empty list with environment variables).

I tried to dig up my clap-rs discussion in their repo but I can't find it anymore.

Reply:
It sounds like a problem that would be solved by not reusing the CLI args DTO for the business logic.

Reply:
> It sounds like a problem that would be solved by not reusing the CLI args DTO for the business logic.

How so? If `clap` doesn't properly support differentiating between passing no argument and passing an empty argument then we still can't "recover" the difference by using more intermediary structs.

Reply:
If we had a separate type for the argument parsing then we could convert into another type that doesn't have the `None` variant. `Vec<ArgEnum>` gets turned into `Vec<DomainEnum>` and the `ArgEnum::None`s are dropped. I can see the appeal of it but won't include it in this PR.